src/Glpi/Api/Rest/ItemHandler.php

```diff
@@ -249,4 +251,55 @@ public function deleteItem($itemType, $id, array $inputValues = [], array $query
       ];
    }
 
+   /**
+    * Magic method to execute CRUD actions with a single ItemTypes dynamically.
+    * Send your ItemType as a method and the arguments to be executed.
+    * Each ItemType's name came from the table's name in the DB schema.
+    * Examples:
+    *    Alerts => glpi_alerts,
+    *    Calendars_Holidays => glpi_calendars_holidays,
+    *    Consumableitems => glpi_consumableitems,
+    *
+    * How to use this?. You can check the file 'example/magicCallExample.php' for reference.
+    *
+    * @param string $name of the itemtype
+    * @param mixed $arguments first is method, second is the input, and third is the query string
+    * @return array|null
+    * @throws InsufficientArgumentsException|BadEndpointException|InvalidArgumentException
+    */
+   public function __call($name, $arguments) {
+      $name = ucfirst($name);
+
+      if (count($arguments) < 2) {
+         throw new InsufficientArgumentsException(ErrorHandler::getMessage('ERROR_APILIB_ARGUMENTS'));
+      }
+
+      $method = strtolower($arguments[0]);
+      $input = isset($arguments[1]) ? $arguments[1] : [];
+      $params = isset($arguments[2]) ? $arguments[2] : [];
+      $result = null;
+      switch ($method) {
+         case 'create':
+            $result = $this->addItem($name, $input);
+            break;
+         case 'read':
+            $result = $this->getAnItem($name, (int)$input, $params);
```

Comment:
getItem() would be closer to the 3 other methods. I think this would be easier to use.

Reply:
Yes